kwave/utils/matlab.py

```diff
@@ -18,16 +43,77 @@ def matlab_find(arr, val=0, mode='neq'):
     return np.expand_dims(arr, -1)  # compatibility, n => [n, 1]
 
 
-def matlab_mask(arr, mask, diff=None):
+def matlab_mask(arr: np.ndarray, mask: np.ndarray, diff: Optional[int] = None) -> np.ndarray:
+    """
+    Applies a mask to an array and returns the masked elements.
+
+    Args:
+        arr: The array to be masked.
+        mask: The mask array, which must be of the same shape as arr.
+        diff: An optional integer to add to the mask indices before applying the mask.
+
+    Returns:
+        A NumPy array containing the masked elements.
+    """
     if diff is None:
         return np.expand_dims(arr.ravel(order='F')[mask.ravel(order='F')], axis=-1)  # compatibility, n => [n, 1]
     else:
         return np.expand_dims(arr.ravel(order='F')[mask.ravel(order='F') + diff], axis=-1)  # compatibility, n => [n, 1]
 
 
-def unflatten_matlab_mask(arr, mask, diff=None):
+def unflatten_matlab_mask(arr: np.ndarray, mask: np.ndarray, diff: Optional[int] = None) -> Tuple[
+    Union[int, np.ndarray], ...]:
+    """
+    Converts a mask array to a tuple of subscript indices for an n-dimensional array.
+
+    Args:
+        arr: The n-dimensional array for which the mask was created.
+        mask: The mask array, which can be of any dimensions.
+        diff: An optional integer to add to the mask indices before converting them to subscript indices.
+
+    Returns:
+        A tuple of integers or NumPy arrays representing the corresponding subscript indices.
+    """
     if diff is None:
         return np.unravel_index(mask.ravel(order='F'), arr.shape, order='F')
     else:
         return np.unravel_index(mask.ravel(order='F') + diff, arr.shape, order='F')
- 
\ No newline at end of file
+
+
+def ind2sub(array_shape: Tuple[int, ...], ind: int) -> Tuple[int, ...]:
```

Comment:
Maybe related to the `conversion` utils? Same question for the function below this.

Reply:
I had thought that as well, but it mirrors a MATLAB method, so I moved it here. If you feel strongly I can move it to conversion.

Reply:
I see, makes sense! Let's keep it here then